packages/cli/src/evaluation.ee/test-runs.controller.ee.ts

```diff
@@ -7,112 +7,91 @@ import { UnexpectedError } from 'n8n-workflow';
 import { ConflictError } from '@/errors/response-errors/conflict.error';
 import { NotFoundError } from '@/errors/response-errors/not-found.error';
 import { NotImplementedError } from '@/errors/response-errors/not-implemented.error';
-import { TestRunsRequest } from '@/evaluation.ee/test-definitions.types.ee';
 import { TestRunnerService } from '@/evaluation.ee/test-runner/test-runner.service.ee';
+import { TestRunsRequest } from '@/evaluation.ee/test-runs.types.ee';
 import { listQueryMiddleware } from '@/middlewares';
 import { getSharedWorkflowIds } from '@/public-api/v1/handlers/workflows/workflows.service';
 import { Telemetry } from '@/telemetry';
+import { WorkflowFinderService } from '@/workflows/workflow-finder.service';
 
-import { TestDefinitionService } from './test-definition.service.ee';
-
-@RestController('/evaluation/test-definitions')
+@RestController('/workflows')
 export class TestRunsController {
 	constructor(
-		private readonly testDefinitionService: TestDefinitionService,
 		private readonly testRunRepository: TestRunRepository,
+		private readonly workflowFinderService: WorkflowFinderService,
 		private readonly testCaseExecutionRepository: TestCaseExecutionRepository,
 		private readonly testRunnerService: TestRunnerService,
 		private readonly instanceSettings: InstanceSettings,
 		private readonly telemetry: Telemetry,
 	) {}
 
-	/**
-	 * This method is used in multiple places in the controller to get the test definition
-	 * (or just check that it exists and the user has access to it).
-	 */
-	private async getTestDefinition(
-		req: TestRunsRequest.GetOne | TestRunsRequest.GetMany | TestRunsRequest.Delete,
-	) {
-		const { testDefinitionId } = req.params;
-
-		const userAccessibleWorkflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
-
-		const testDefinition = await this.testDefinitionService.findOne(
-			testDefinitionId,
-			userAccessibleWorkflowIds,
-		);
-
-		if (!testDefinition) throw new NotFoundError('Test definition not found');
-
-		return testDefinition;
-	}
-
 	/**
 	 * Get the test run (or just check that it exists and the user has access to it)
 	 */
 	private async getTestRun(
 		req: TestRunsRequest.GetOne | TestRunsRequest.Delete | TestRunsRequest.Cancel,
 	) {
-		const { id: testRunId, testDefinitionId } = req.params;
+		const { id: testRunId, workflowId } = req.params;
+
+		const sharedWorkflowsIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		if (!sharedWorkflowsIds.length || (workflowId && !sharedWorkflowsIds.includes(workflowId))) {
```

Comment:
Previous comments could help simplifying this with only: `!sharedWorkflowsIds.includes(workflowId)` I think